new-frontend/src/app/guards/chosenCollaboration.guard.ts

```diff
@@ -2,16 +2,23 @@ import { inject } from '@angular/core';
 import { CanActivateFn, Router } from '@angular/router';
 import { routePaths } from '../routes';
 import { CHOSEN_COLLABORATION } from '../models/constants/sessionStorage';
+import { ResourceType, OperationType } from '../models/api/rule.model';
+import { PermissionService } from '../services/permission.service';
+import { CollaborationService } from '../services/collaboration.service';
 
-export function chosenCollaborationGuard(): CanActivateFn {
+export function chosenCollaborationGuard(resource?: ResourceType, operation?: OperationType): CanActivateFn {
```

Comment:
This guard has only one purpose, checking if there is a chosenCollaboration the user is allowed to use. It does not need resource and operation as a parameter, as these values will always be the same.

Reply:
I don't think that's true: this collaboration guard can be used for different pages. In the PR, it is used in to check if someone is allowed to access a page where you can create a task for that collaboration:
```
        path: routerConfig.taskCreate,
        component: TaskCreateComponent,
        canActivate: [authenticationGuard(), chosenCollaborationGuard(ResourceType.TASK, OperationType.CREATE)]
```
This guard may also be used to access a page where you view all the tasks of that collaboration, or where you edit/delete the collaboration itself

Reply:
I would expect the chosenCollaborationGuard to only check if there is a chosen collaboration. I than would expect a new authorizationGuard to check if the user has the right roles. So the chosenCollaborationGuard can be used on pages that need a chosen collaboration (for example the whole admin section doesn't that) and the authorizationGuard can be used for all pages that need a specific role (or multiple).